### PR TITLE
bpo-36096: IDLE: Refactor class variables in colorizer

### PR DESCRIPTION
**Pull request by csabella**
Refactor class variables to instance variables in `colorizer.py`.

https://bugs.python.org/issue36096

### CODE REVIEW

**Review by eamanu — APPROVED** [review]

File: Lib/idlelib/colorizer.py
Comment (terryjreedy) [inline Lib/idlelib/colorizer.py:70]: close_when_done is not used by turtledemo.__main__, so we can get rid of it.  I want to do this in its own issue.

File: Misc/NEWS.d/next/IDLE/2019-02-23-17-53-53.bpo-36096.mN5Ly3.rst
Comment (terryjreedy) [inline Misc/NEWS.d/next/IDLE/2019-02-23-17-53-53.bpo-36096.mN5Ly3.rst:1]: For the present, I will take care of the idlelib news.txt entries.

**Review by terryjreedy — APPROVED** [review]

File: Misc/NEWS.d/next/IDLE/2019-02-23-17-53-53.bpo-36096.mN5Ly3.rst
Comment (csabella) [reply Misc/NEWS.d/next/IDLE/2019-02-23-17-53-53.bpo-36096.mN5Ly3.rst:1]: I should have asked this earlier, but once you approve a PR, should I then merge it?  Thanks.

Comment (terryjreedy) [reply Misc/NEWS.d/next/IDLE/2019-02-23-17-53-53.bpo-36096.mN5Ly3.rst:1]: If I check the []Approve box for a Review, and if you agree that it is ready, and accept the responsibility of doing the merge, yes.  For most IDLE issues though, I consider the followup responsibility to be shared.  If you prefer me to merge, say so.  If a PR is actually a joint product, or if it is blocking doing another, I might merge it, but should say so first.

Comment (csabella) [reply Misc/NEWS.d/next/IDLE/2019-02-23-17-53-53.bpo-36096.mN5Ly3.rst:1]: Thank you, that's very helpful.  I think this one looks OK and you've approved it, so I'll merge it soon.

**Comment by terryjreedy** [comment]
As will nearly always be the case, these post-merge buildbot failure reports are 'false positives' in relation to the merge.  Both are failures in test_urllib2net.

Looking at https://buildbot.python.org/all/#/console, it appears that the two merges before this broke almost everything.  Re-adding the backport label has not triggered another backport after 10+ minutes, so I suspect the bot is disabled.

**Comment by csabella** [comment]
You added the 'need backport to 3.7' label to the closed PR and it started the backporting process all over again?  That's a good trick to know about!

**Pull request merged**
